src/StreamJsonRpc.Shared/DataContracts/JsonRpcMessage.cs

```diff
@@ -67,7 +67,14 @@ public static JsonRpcMessage CreateRequest(int? id, string @method, IReadOnlyLis
 
         public static JsonRpcMessage CreateRequestWithNamedParameters(int? id, string @method, object namedParameters, JsonSerializer parameterSerializer)
         {
-            return new JsonRpcMessage(method, JToken.FromObject(namedParameters, parameterSerializer), id);
+            if (namedParameters == null)
```

Comment:
Is this legal? The json-rpc 2.0 spec reads: 

> by-name: params MUST be an Object

No allowance is made for a null value here. So it seems to me that we should pass `{}` when no parameters are specified, or (perhaps better) throw an exception so folks realize they must pass in an empty object, since that's what the remote end will get.